### ISSUE TITLE
Application generator should create app with declaratively loaded loopback.rest and explorer

### ISSUE THREAD

**Issue by crandmck** [strongloop/generator-loopback#113]
Recent changes (e.g. https://github.com/strongloop/loopback-boot/pull/149 and https://github.com/strongloop/loopback-example-app/pull/77) enable an app to load loopback.rest declaratively in `middleware.json` and loopback.explorer via `component-config.json`.
The application generator needs to reflect this.  

So by default, the generated `middleware.json` should have the following to load REST and status middleware:
```
"routes": {
    "loopback#rest": {
      "paths": ["${restApiRoot}"]
    },
    "loopback#status" : {
      "paths" : ["/"]
    }
  },
...
```

**NOTE**: I'm not 100% sure about the status middleware config above.  @bajtos can you confirm that is correct?

The above replaces `server/boot/rest-api.js` and `server/boot/root.js` which are no longer needed AFAIK.

Also, the generator should create a `server/component-config.json` that looks like this:
```
{
  "loopback-explorer": {
    "mountPath": "/explorer"
  }
}
```
The above replaces `server/boot/explorer.js`, which is no longer needed AFAIK.

**Comment by crandmck** [comment]
@bajtos suggested that @hacksparrow could handle this fix, but I know Al wants him to focus on the Express generator work.  I have assigned to @chandadharap to sort this out.

**Comment by bajtos** [comment]
We need to change `server/server.js` too. The pull request https://github.com/strongloop/loopback-example-app/pull/77 contains all changes we need to make. loopback-example-app should contain exactly the output of `slc loopback:example`.

**Comment by hacksparrow** [comment]
An issue in `loopback-workspace` - https://github.com/strongloop/loopback-workspace/issues/228 - is holding back the update to the generator.

**Comment by crandmck** [comment]
Looks like you can workaround that issue...let me know if there are any other questions or issues.

**Comment by hacksparrow** [comment]
https://github.com/strongloop/loopback-workspace/pull/230 addresses this issue in whole.

Opened a PR, https://github.com/strongloop/generator-loopback/pull/114, updating a unit test to ensure the changes introduced by https://github.com/strongloop/loopback-workspace/pull/230 are taken into consideration.

**Comment by hacksparrow** [comment]
@crandmck @bajtos I think we can close this now, fixed in `loopback-workspace@3.18.0`.

**Comment by bajtos** [comment]
> I think we can close this now, fixed in loopback-workspace@3.18.0.

:+1:
